packages/steps/src/BuildStepInput.ts

```diff
@@ -129,7 +130,23 @@ export class BuildStepInput<
         valueInterpolatedWithGlobalContext,
         (path) => this.ctx.getStepOutputValue(path) ?? ''
       );
-      return this.parseInputValueToAllowedType(valueInterpolatedWithOutputsAndGlobalContext);
+      returnValue = this.parseInputValueToAllowedType(valueInterpolatedWithOutputsAndGlobalContext);
+    }
+    return this.fixEscapeCharacters(returnValue);
+  }
+
+  private fixEscapeCharacters(
+    input: BuildStepInputValueTypeWithRequired<T, R>
+  ): BuildStepInputValueTypeWithRequired<T, R> {
+    if (typeof input === 'string') {
+      return input.replace(/\\n/g, '\n') as BuildStepInputValueTypeWithRequired<T, R>;
```

Comment:
There is `decodeURI[Component]` which could decode these. However, a bigger question is why do these strings contain `\\n` and not `\n`? Maybe we don't need to decode input, but instead we should fix output?

Reply:
I agree, but atm I don't know if the problem is the earlier step setting the output, reading the outputs value, interpolating the input with other steps output, reading the interpolated input, or any other place.
I propose releasing this, since it actually fixes the problem that the user has and then digging deeper to look for the root cause